### DISCUSSION TITLE
以UI顺序获取控件的值 / Get Form Values By Order In UI

### DISCUSSION THREAD

**Discussion by zhenchuan9r** [alibaba/formily#3806] [Q&A 2.x]
[https://core.formilyjs.org/api/models/form](Formily-Core/Form)
We need a method to get form values by order in accordance to UI layout.
For example, if Component A is on the upper side of Component B, the position of value of Component A should come before that of Component B, like:
`
{
"component-a": {},
"component-b":{}
}
`
I would like to ask if there is already such kind of method or we have to parse and traverse the config of the forms.

**Comment by zhenchuan9r** [comment]
[https://core.formilyjs.org/api/models/form](https://github.com/alibaba/formily/discussions/Formily-Core/Form)
我们需要一种按照 UI 布局顺序获取表单值的方法。
例如，如果组件 A 位于组件 B 的上方，则组件 A 的值的位置应该在组件 B 的值之前，如下所示：
{ "component-a": {}, "component-b":{} }
想问一下是否已经有这样的方法，或者我们必须解析和遍历表单的配置。

**Reply by janryWang** [reply]
object 顺序，这个没法基于布局来定的
